conwayste/src/client.rs

```diff
@@ -852,17 +859,25 @@ impl MainState {
 
     fn pause_or_resume_game(&mut self) {
         let cur_menu_state = self.menu_sys.menu_state;
-        let cur_stage = self.stage;
+        let current_screen = match self.screen_stack.last() {
+            Some(screen) => screen,
+            None => panic!("Error in key_down_event! Screen_stack is empty!"),
+        };
 
-        match cur_stage {
-            Stage::Menu => {
+        match current_screen {
+            Screen::Menu => {
                 if cur_menu_state == menu::MenuState::MainMenu {
-                    self.stage = Stage::Run;
+                    // If at 1, then we haven't started the game yet
```

Comment:
This feels a little hacky to me.

Reply:
Agreed. Would you be opposed if peeked at the screen layer below to see if it was `Screen::Menu` ?